### DISCUSSION TITLE
contentContainerStyle flexGrow: 1 to allow center ListEmptyComponent

### DISCUSSION THREAD

**Discussion by tomasmozeris** [Shopify/flash-list#517] [General]
**Problem**
I have `ListEmptyComponent` component, which uses style `flex: 1, justifyContent: 'center'` to place it inside center of users viewport. 
https://snack.expo.dev/7MSfUAke7

**My workaround**
To allow empty list container take remaining space I had to patch `flash-list` and `recyclerlistview` with `flexGrow: 1` in these places:
https://github.com/Shopify/flash-list/blob/main/src/FlashList.tsx#L327
https://github.com/Flipkart/recyclerlistview/blob/master/src/platform/reactnative/scrollcomponent/ScrollComponent.tsx#L86

As default `<Flatlist />` allows `contentContainerStyle={{ flexGrow: 1 }}` and solves such problem, consider to support `flexGrow` in `contentContainerStyle of Flashlist`

**Comment by Poheart** [comment]
Would like to bump this as it deserve more attention.

i have the same use case with yours on ListEmptyComponent, and need to apply `flexGrow` to fill the available space on screen.

**Comment by ser-emejia** [comment]
Please, also add the 

_**columnWrapperStyle**_ prop

When we have two or more columns in our layout

**Comment by Navipro70** [comment]
I think issue is more efficient to point the problem, unfortunately

**Comment by ansh** [comment]
This is a big issue that should be fixed!

**Comment by Bayramito** [comment]
i think you could use CellRendererComponent for that

**Comment by chj-damon** [comment]
I just encountered the same problem. I want to center my ListEmptyComponent

**Reply by ser-emejia** [reply]
Hello, this solution works for me:

```
const [height, setHeight] = useState<number>(0);

<View
    style={{ flex:1 }}
    onLayout={(event) => setHeight(event.nativeEvent.layout.height)}>
    <FlashList
        ...
        ListEmptyComponent={
            <View
                style={{
                    height,
                    alignItems: 'center',
                    justifyContent: 'center' 
                }}>
                <Text>Text Center</Text>
            </View>
        }
    />
</View>
```

Hope it helps!

**Reply by chj-damon** [reply]
yes it works. but it feels not natural.

**Reply by NikhilMasurkar** [reply]
how can we add custom styling in this contentContainerStyle?
like this 
`contentContainerStyle={[
              styles.containerStyle,
              customContainerStyle,
            ]}`

**Reply by LukyLurks** [reply]
Thanks for this. Just a few notes from trying it:
- it's a bit messier if you use a HeaderComponent: its height will be included in `event.nativeEvent.layout.height`, and your `ListEmptyComponent` will end up being taller than desired. I didn't thoroughly test it but one can probably work around this by applying the same technique to the header to get its height, then subtract that from the total height. For instance:
```jsx
const [totalHeight, setTotalHeight] = useState<number>(0);
const [headerHeight, setHeaderHeight] = useState<number>(0);

<View
    style={{ flex:1 }}
    onLayout={(event) => setTotalHeight(event.nativeEvent.layout.height)}>
    <FlashList
        /*...*/
        HeaderComponent={() => (
            <View onLayout={(event) => setHeaderHeight(event.nativeEvent.layout.height)}>
                <MyHeaderComponent />
            </View>
        )}
        ListEmptyComponent={
            <View
                style={{
                    height: totalHeight - headerHeight,
                    ...styleThatCentersMyEmptyComponent
                }}>
                <MyEmptyComponent />
            </View>
        }
    />
</View>
```
I couldn't use that because my header has a "Read More" component in it that changes its height, and this workaround causes it to malfunction.

If your header component has a known constant height it can work, but then you can also just subtract that known height from the total height, without getting it through `onLayout`.

- Surely people thought about it but I haven't seen it mentioned. Another workaround, if applicable, could be a simple condition. For instance:
```jsx
return !data || data.length === 0 ? (
    <View style={styleThatCentersMyEmptyComponent}>
        <MyEmptyComponent />
    </View>
) : (
    <FlashList
	/* your usual props */
    />
)
```

**Comment by zacdemi** [comment]
I also need `flexGrow` on contentContainerStyle property render items from the top of an inverted Flashlist.

**Comment by jcalem** [comment]
+1 I also need this added

**Comment by baesumin** [comment]
+1

**Comment by chj-damon** [comment]
@naqvitalha does this worth to take a look?

**Comment by andreialecu** [comment]
We recently merged Flashlist support to [react-native-collapsible-tab-view](https://github.com/PedroBern/react-native-collapsible-tab-view) and ran into a similar isssue:
https://github.com/PedroBern/react-native-collapsible-tab-view/issues/335

We were using `minHeight` previously, but I think `flexGrow` would also work.

**Comment by victoriomolina** [comment]
any updates on this?

**Comment by nazmeln** [comment]
any updates on this one? 🙂

**Comment by usmansbk** [comment]
damn! not fixed

**Comment by hadnet** [comment]
F**, I'm facing this issue and the @tomasmozeris does not work anymore in v1.5.

**Comment by versten1uk** [comment]
What a complex bug to fix, this may be a blocker to migrate from FlatList to FlashList

**Comment by sergey-king** [comment]
Any updates on this?

**Comment by Saschamz** [comment]
+1

**Comment by ameer2468** [comment]
+1

A workaround for now is just to conditionally render.

**Comment by flodlc** [comment]
Hope it will be supported soon

**Comment by amjadbouhouch** [comment]
+1

**Comment by szarbartosz** [comment]
addressed in #1206

**Comment by Samide47** [comment]
same problem using FlashList 1.5.0

**Comment by angelopedroso** [comment]
For those that use nativewind, just add "my-auto" on "contentContainerClassName":

```
contentContainerClassName="my-auto"
```

or create a CSS class:

```
.my-auto {
  margin-top: auto;
  margin-bottom: auto;
}
```

and add on contentContainerClassName

**Comment by tomisykora** [comment]
I was able to overcome this by adding `minHeight: 100%` to ListEmptyComponent's style prop and by adding the following to FlashList props:
```
overrideProps={{
    contentContainerStyle: { flexGrow: 1 }
}}
```
I'm using FlashList v1.7.2.

**Reply by tiempham** [reply]
Thank Guy!

**Comment by hanwenbo** [comment]
```
 overrideProps={{
      contentContainerStyle: { flexGrow: 1,backgroundColor:"green" }
}}

ListEmptyComponent height = 100%
```

**Reply by burhanyilmaz** [reply]
This is worked for my implementation :) Thank you!

**Comment by devlasbe** [comment]
update please

**Comment by RenzLesterH** [comment]
same issue here
